pkg/asset/installconfig/platform.go

```diff
@@ -60,30 +58,59 @@ func (a *Platform) Name() string {
 }
 
 func (a *Platform) queryUserForPlatform() string {
-	for {
-		input := asset.QueryUser(a.InputReader, platformPrompt)
-		input = strings.ToLower(input)
-		for _, p := range validPlatforms {
-			if input == p {
-				return p
-			}
-		}
-		fmt.Println("Invalid platform")
-	}
+	var platform string
+	survey.AskOne(&survey.Select{
+		Message: "Platform",
+		Options: validPlatforms,
+	}, &platform, nil)
+
+	return platform
 }
 
 func (a *Platform) awsPlatform() (*asset.State, error) {
+	var region string
+	survey.AskOne(&survey.Select{
+		Message: "Region",
+		Help:    "The AWS region to be used for installation.",
+		Default: "us-east-1 (N. Virginia)",
+		Options: []string{
+			"us-east-2 (Ohio)",
```

Comment:
Did you work this up by hand?  I'd rather auto-generate it.  You can get close with:

```console
$ aws ec2 describe-regions --query Regions --output json | jq '[.[] | .RegionName] | sort'
[
  "ap-northeast-1",
  "ap-northeast-2",
  "ap-south-1",
  "ap-southeast-1",
  "ap-southeast-2",
  "ca-central-1",
  "eu-central-1",
  "eu-west-1",
  "eu-west-2",
  "eu-west-3",
  "sa-east-1",
  "us-east-1",
  "us-east-2",
  "us-west-1",
  "us-west-2"
]
```

But I can't find a way to get the human names ("Ohio" here).  There's more [here][1] (including "[how do you get the human name?][2]").  Amazon provides [this][3].

The CLI query also doesn't list GovCloud regions (presumably because I don't have access to any of those).  And [the AWS page][3] talks about plans for additional future regions.  So maybe just throw up our hands and don't list options?

[1]: https://gist.github.com/neilstuartcraig/0ccefcf0887f29b7f240
[2]: https://gist.github.com/neilstuartcraig/0ccefcf0887f29b7f240#gistcomment-2599847
[3]: https://aws.amazon.com/about-aws/global-infrastructure/

Reply:
I copied the list from AWS's documentation.

My thinking with the regions was that if the user couldn't find their region, they could just modify install-config.yaml after the fact. This list won't include GovCloud or AWS China.